Model/Credit_Card_Payment_Method.php

```diff
@@ -0,0 +1,131 @@
+
+use \OxidEsales\Eshop\Application\Model\Payment;
+use \OxidEsales\Eshop\Core\Registry;
+
+use Wirecard\Oxid\Core\Helper;
+
+/**
+ * Payment method implementation for Credit Card
+ */
+class Credit_Card_Payment_Method extends Payment_Method
+{
+    /**
+     * @inheritdoc
+     */
+    protected static $_sName = "creditcard";
+
+    /**
+     * Get the payment method's configuration
+     *
+     * @return Config
+     *
+     * @SuppressWarnings(PHPMD.Coverage)
+     */
+    public function getConfig()
+    {
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load(self::getName(true));
+        $oConfig = new Config(
+            $oPayment->oxpayments__wdoxidee_apiurl->value,
+            $oPayment->oxpayments__wdoxidee_httpuser->value,
+            $oPayment->oxpayments__wdoxidee_httppass->value
+        );
+        $oPaymentMethodConfig = new PaymentMethodConfig(
+            self::getName(),
+            $oPayment->oxpayments__wdoxidee_maid->value,
+            $oPayment->oxpayments__wdoxidee_secret->value
+        );
+        $oConfig->add($oPaymentMethodConfig);
+
+        return $oConfig;
+    }
+
+    /**
+     * Get the current transaction to be processed
+     *
+     * @return Transaction
+     *
+     * @SuppressWarnings(PHPMD.Coverage)
+     */
+    public function getTransaction()
+    {
+        return new CreditCardTransaction();
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     */
+    public function getConfigFields()
+    {
+        $parentConfigFields = parent::getConfigFields();
+        $additionalFields = [
+            'threeDMaid' => [
+                'type'        => 'text',
+                'field'       => 'oxpayments__wdoxidee_three_d_maid',
+                'title'       => Helper::translate('config_three_d_merchant_account_id'),
+                'description' => Helper::translate('config_three_d_merchant_account_id_desc'),
+            ],
+            'threeDSecret' => [
+                'type'        => 'text',
+                'field'       => 'oxpayments__wdoxidee_three_d_secret',
+                'title'       => Helper::translate('config_three_d_merchant_secret'),
+                'description' => Helper::translate('config_three_d_merchant_secret_desc'),
+            ],
+            'nonThreeDMaxLimit' => [
+                'type'        => 'text',
+                'field'       => 'oxpayments__wdoxidee_non_three_d_max_limit',
+                'title'       => Helper::translate('config_ssl_max_limit'),
+                'description' => Helper::translate('config_ssl_max_limit_desc'),
+            ],
+            'threeDMinLimit' => [
+                'type'        => 'text',
+                'field'       => 'oxpayments__wdoxidee_three_d_min_limit',
+                'title'       => Helper::translate('config_three_d_min_limit'),
+                'description' => Helper::translate('config_three_d_min_limit_desc'),
+            ],
+            'basket' => [
```

Comment:
I think basket is the wrong key for the currency

Reply:
fixed key naming